src/rust/src/decoder/mod.rs

```diff
@@ -86,10 +88,161 @@ impl dtvcc_ctx {
             ),
         }
     }
-    pub fn process_current_packet(&mut self, len: i32) {
+    /// Process current packet into service blocks
+    pub fn process_current_packet(&mut self, len: u8) {
+        let seq = (self.current_packet[0] & 0xC0) >> 6;
+        debug!(
+            "dtvcc_process_current_packet: Sequence: {}, packet length: {}",
+            seq, len
+        );
+        if self.current_packet_length == 0 {
+            return;
+        }
+
+        if self.current_packet_length != len as i32 {
+            // Is this possible?
+            self.reset_decoders();
+            return;
+        }
+
+        if self.last_sequence != CCX_DTVCC_NO_LAST_SEQUENCE
+            && (self.last_sequence + 1) % 4 != seq as i32
+        {
+            warn!("dtvcc_process_current_packet: Unexpected sequence number, it is {} but should be {}", seq, (self.last_sequence +1) % 4);
+        }
+        self.last_sequence = seq as i32;
```

Comment:
Are all these type conversions/declarations really needed?

Reply:
Yes, because the C bindings use `c_int`, which is `i32` in rust